advance_guide/imp_websockets/main.py

```
@@ -0,0 +1,90 @@
+
+    async def send_personal_message(self, message:str, websocket: WebSocket):
+        await websocket.send_text(message)
+
+    async def broadcast(self, message: str):
+        for connection in self.active_connections:
+            await connection.send_text(message)
+
+manager = ConnectionManager()
+
+@app.websocket("/ws/{client_id}")
+async def websocket_endpoint(websocket: WebSocket, client_id: int):
```

Ruff (0.8.2)

Redefinition of unused `websocket_endpoint` from line 23

(F811)